### PR TITLE
Modify Crop and Soil outputs for compatilibility with EEE

### LINKED ISSUE #1005: Add tillage implements input to the tillage application schedule input
**Is your feature request related to a problem? Please describe.**
In order for the Energy and Emissions module to properly estimate outputs related to tillage practices, it needs to be know what implement was used to execute each tillage operation.

**Describe the solution you'd like**
- An array of strings should be added to the tillage schedule input file, with each string being the name of the tillage implement used.
- The `TillageEvent` and `TillageSchedule` classes should properly handle this additional input.
- When the `TillageApplication` class executes a tillage event, it should record the implement used in addition to the other tillage event information recorded.

**Describe alternatives you've considered**
Have the Energy and Emissions module make a best guess at what implement was used to execute the tillage operation based on the operation's parameters (depth, mixing efficiency and incorporation fraction).

**Additional context**
As of 11/29, there is not a set list of tillage implements that are supported for V1. Roohi is working on creating a mapping between implements that are supported in the Energy and Emissions module, and the implements listed in the SWAT tillage database ([table A-9](https://swat.tamu.edu/media/69419/Appendix-A.pdf)).

### PR DESCRIPTION
**Pull request by Isgotkowitz**
Modifies the Crop and Soil module so that field operations related outputs can be used by the EEE module to predict energy use. The 5 outputs that have been modified are the recordings of planting, harvesting, tilling, manure applications, and fertilizer applications.

The most significant change is to tillage operations. The enum `TillageImplement` has been introduced in the EEE module, and is used by the Crop and Soil module to record which implement "executed" a tillage operation. This implement must be specified by the user, the same as tillage depth, mixing fraction, etc.

Plantings and harvestings indicate the crop planted/harvested with the `CropSpecies` enum. This enum has been moved to its own file.

All recorded events include the size of the field that the operation was performed on, all events except harvestings include the percentage of clay in the soil, and all events except harvestings and plantings include how deep they went into the soil.

Closes #1005 

To pull the information needed by EEE out of the OM (for Crop and Soil related estimations) this filter can be used:
```json
{
  "multiple": [
    {
      "name": "Fertilizer",
      "filters": ["Field._record_fertilizer_application.fertilizer_application.field='.*'"],
      "variables": ["mass", "application_depth", "field_size", "average_clay_percent"]
    },
    {
      "name": "Tillage",
      "filters": ["TillageApplication._record_tillage.tillage_record.field='.*'"],
      "variables": ["tillage_depth", "implement", "field_size", "average_clay_percent"]
    },
    {
      "name": "Manure",
      "filters": ["Field._record_manure_application.manure_application.field='.*'"],
      "variables": ["dry_matter_mass", "dry_matter_fraction", "application_depth", "field_size", "average_clay_percent"]
    },
    {
      "name": "Harvestings",
      "filters": ["CropManagement._record_yield.harvest_yield.field='.*'"],
      "variables": ["dry_yield", "crop", "field_size"]
    },
    {
      "name": "Plantings",
      "filters": ["Field._plant_crop.crop_planting.field='.*'"],
      "variables": ["crop", "field_size", "average_clay_percent"]
    }
  ]
}
```

### CODE REVIEW

File: RUFAS/routines/EEE/enums.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    MOLDBOARD_PLOW = "moldboard-plow"
+    COULTER_CHISEL_PLOW = "coulter-chisel-plow"
+    DISK_HARROW = "disk-harrow"
+    CULTIVATOR = "cultivator"
+    SEEDBED_CONDITIONER = "seedbed-conditioner"
+
+    def __str__(self):
+        return self.value
+
+    def __repr__(self):
```
Comment (PooyaHekmati) [inline RUFAS/routines/EEE/enums.py]: I see the same pattern in CropSpecies as well. Overriding repr with value is not a good practice. Unless it is absolutely necessary, avoid doing so.

**Review by PooyaHekmati — APPROVED** [review]

File: RUFAS/routines/EEE/enums.py (hunk 1 above)
Comment (Isgotkowitz) [reply RUFAS/routines/EEE/enums.py]: Ok. I will look into it and see if I can figure out a way around it.

**Comment by PooyaHekmati** [comment]
For record keeping purposes, can you share the corresponding filter here?

**Comment by Isgotkowitz** [comment]
To pull the information needed by EEE out of the OM (for Crop and Soil related estimations) this filter can be used:
```json
{
  "multiple": [
    {
      "name": "Fertilizer",
      "filters": ["Field._record_fertilizer_application.fertilizer_application.field='.*'"],
      "variables": ["mass", "application_depth", "field_size", "average_clay_percent"]
    },
    {
      "name": "Tillage",
      "filters": ["TillageApplication._record_tillage.tillage_record.field='.*'"],
      "variables": ["tillage_depth", "implement", "field_size", "average_clay_percent"]
    },
    {
      "name": "Manure",
      "filters": ["Field._record_manure_application.manure_application.field='.*'"],
      "variables": ["dry_matter_mass", "dry_matter_fraction", "application_depth", "field_size", "average_clay_percent"]
    },
    {
      "name": "Harvestings",
      "filters": ["CropManagement._record_yield.harvest_yield.field='.*'"],
      "variables": ["dry_yield", "crop", "field_size"]
    },
    {
      "name": "Plantings",
      "filters": ["Field._plant_crop.crop_planting.field='.*'"],
      "variables": ["crop", "field_size", "average_clay_percent"]
    }
  ]
}
```

File: RUFAS/routines/EEE/enums.py (hunk 1 above)
Comment (Isgotkowitz) [reply RUFAS/routines/EEE/enums.py]: Found a workaround, removed the overriding of repr. Thanks for pointing this out Pooya.

**Review by KFosterReed — APPROVED** [review]
This looks good to me and produces the outputs as expected!

**Pull request merged**